Pulling fresh changes from prebid/Prebid.js

pawel-lankocz opened the pull request:
Pulling fresh changes from prebid/Prebid.js

lm-abhijit committed:
Lemma:set mediaType key value (#6006)

* lemmaBidAdapter.js

Added lemma bid adapter file

* lemmaBidAdapter.md

Added lemma bid adapter md file

* lemmaBidAdapter_spec.js

Added lemma bid adapter test spec file

* Update lemmaBidAdapter.js

Fixed automated code review alert comparison between inconvertible types

* Update lemmaBidAdapter.js

Fixed review changes

* Update lemmaBidAdapter.md

Correct parameter value.

* Update lemmaBidAdapter.js

Lemma Bid Adapter - v3.0 compliance

* Update lemmaBidAdapter_spec.js

Lemma Bid Adapter - v3.0 compliance

* Update lemmaBidAdapter.md

Lemma Bid Adapter - v3.0 compliance

* Update lemmaBidAdapter.js

Added user sync support into bid adapter.

* updated include modules file extension.

updated include modules js file extension.

* Update lemmaBidAdapter_spec.js

Added unit test for user sync feature.

* Update lemmaBidAdapter.js

Fixed format error.

* Update lemmaBidAdapter_spec.js

Fixed format error and typo error.

* Set mediaType key value into bid object

Set mediaType key value into the bid object.

* Update lemmaBidAdapter.js

remove duplicate function

* Update lemmaBidAdapter.js

Remove non supported code.

* Update lemmaBidAdapter_spec.js

Remove GDPR test cases.

ix-certification committed:
added support for addtlConsent (#6005)

Co-authored-by: Ix-Prebid-Support <ix-prebid-support@indexexchange.com>

dev-missena committed:
Add bid adapter for Missena (#6247)

* adds support for getFloor of video mediaTypes

* adds test for calling getFloor with correct mediaType

* checks that _getFloor converts string floors to float

* Add bid adapter for Missena

* Use publisher demo token in tests

* Add Missena global vendor ID to spec

* Use apiKey in the current bidRequest

* Add referer info to payload

Co-authored-by: Nick Llerandi <nllerandi@triplelift.com>
Co-authored-by: Brandon Ling <51931757+blingster7@users.noreply.github.com>
Co-authored-by: Brandon Ling <bling@triplelift.com>

pm-harshad-mane committed:
Gulp test file example added in readme (#6287)

poojawins committed:
Sharethrough: Add support for ID5, Shared ID, and Live Intent ID (#6261)

* Update prebid adapter universal ids to include ID5, SharedID, and LiveIntent ID.

[#176447070](https://www.pivotaltracker.com/story/show/176447070)

Co-authored-by: Mathieu Pheulpin <mpheulpin@sharethrough.com>

* Addressing review

[#176447070]

* Quick rewrite

[#176447070]

* Address ID5 review, forward linkType to adserver

* Reformatting SharedID to align with ID5

Co-authored-by: Mathieu Pheulpin <mpheulpin@sharethrough.com>

lowendavid committed:
SmartAdServer Bid Adapter: image sync and noAd (#6236)

* SIM-889 Now we have image based sync

* SIM-889 Added test to check noad and image sync

* SIM-889 Fixing indenting issues

iskmerof committed:
Add client Alias Adkernel (#6291)

Adding "bcm" alias to Adkernel adapter

icflournoy committed:
[ParrableIdSystem] Add GVLID and handle TC Consent data (#6283)

* Added GDPR support

* Remove forgotten .only

Co-authored-by: Victor <victorigualada@gmail.com>

mkendall07 committed:
4.26.0 release

mkendall07 committed:
4.27.0-pre

Commit:
Updated data mapping of winning bid and auction logs in pubxai analytics adapter (#6285)

Co-authored-by: Phaneendra Hegde <phaneendra.hegde@gmail.com>

TheMediaGrid committed:
Grid Bid Adapter: Added video protocols to the ad request (#6299)

bretg committed:
Rubicon Bid Adapter: updated transactionId to auctionId for OpenRTB (#6298)

pm-harshad-mane committed:
Fix for Issue 6117: Added Module Name in Build to Comments (#6297)

patmmccann committed:
map tripleliftBidAdapter.js tl_source to bid.meta.mediaType (#6303)

prebidtappx committed:
Tappx Bid Adapter: new bidder adapter added (#6233)

* ADD: tappx bid adapter

* FIX: replace .includes by .indexOf

* UPDATE: Expand the test coverage

* FIX: format spacing tests

* FIX: get auctionId from validBidRequests

* UPDATE: add bannerMediaType tablet sizes

* FIX: get timeout from bidderRequest.timeout

* UPDATE: replace the way to get the hostname

* UPDATE: adding  support multiple bid requests in a single call

* UPDATE: remove hardcoded test payload param

Co-authored-by: marc_tappx <marc@tappx.com>

guiann committed:
Add Native ad capability to AdYouLike adapter (#6198)

* Remove useless bidderCode in bid response

* send all the available sizes in the bid request

* Use the banner sizes if given

* avoid compatibility issue with old bid format

* ad iframe and publisher domain paramters to bid requests

* add publisher domain info in ad request

* add a check in unit tests for publisherDomain

* encode uri components

* add native assets support

* add native information to ad request

* fix native ad parsing

* fix nativ condition to set mediatype

* fix image access and add trackers

* fix and add unit test on native ad

* update md file

* remove usage of URLSearchParams

* allox pure native ad with no adm provided

Co-authored-by: Guillaume <guiandouard@gmail.com>

patmmccann committed:
Update targeting.js to fix TTL buffer arithmetic (#6300)

dreischer committed:
Permutive RTD submodule (#6290)

* Add Permutive RTD module

* set demo data in LS for test page

* fix linter issues

* reduce timeout on example page

* decrease sample timeouts

* rename targeting to segments

GeneGenie committed:
Adtelligent Bid Adapter: Add Navelix as alias (#6306)

* Navelix bidder

* gvlID

Commit:
Pubmatic add warning for non-string adSlots (#6304)

* add warning for non-string adSlots

- and update the markdown file to say they should be strings

* be more specific in explanatory comment

stevealliance committed:
Update ttl outstream (#6253)

* adding DMX

test @97%, two files added one updated

* Update districtm_spec.js

* Update districtmDMX.js

* adding all districtm needed file

* remove legacy file

* remove typo || 0 in the test method

* force default to return a valid width and height

* update unit test code for failing test

* changed class for an object

* remove package-lock.json

* change file name for dmx adapter

* renamed files

* restaure package-lock.json

* update to last package-lock state

* update gdpr user consent

* fix sizes issue

* Documentation updates

Adding the readme.md info

* update file name and update unit testing import file location

* current machine state

* lint correction

* remove variable assigment duplicate

* adding CCPA support for DMX

* adding test for ccpa and gdpr

* districtm dmx adding deal id field

* idsync support ccpa & gdpr

* merge forked

* stop mediatype changing ttl

* remove lint error

* adding gvlid for dmx

* removing change

Removing change from on commit PR file package-lock.json

* update file 

Removing comments on DMX adapter

* Revert "removing change"

This reverts commit e4bbbe55

* adding aliases base on ticket #6294

* update file 

Removing comments on DMX adapter

* fix error for issue #6294

* file removal from PR

Co-authored-by: Steve Alliance <steve@districtm.ca>
Co-authored-by: Luis <luissastreverzun@gmail.com>
Co-authored-by: Steve Alliance <stevealliance@Steves-Air.localdomain>
Co-authored-by: Steve Alliance <stevealliance@Steves-MacBook-Air.local>
Co-authored-by: steve-a-districtm <steve@districtm.net>
Co-authored-by: Steve Alliance <stevealliance@DMMBP-C02WXCFAJG5H.local>

icflournoy committed:
[ParrableIdSystem] Accept lowercase timezone names in timezone filter (#6282)

* Add lower case comparison for allowedZones

* Lowercase blocked zones also

* Fixed timezone lowercase tests

* Handle IE missing timeZone field

* Improve lowercase timezone comparison

Co-authored-by: Victor <victorigualada@gmail.com>

thomas-netric committed:
Added new size - Id 550 (980x480) (#6311)

bretg committed:
Revert "Rubicon Bid Adapter: updated transactionId to auctionId for OpenRTB (#6298)" (#6314)

This reverts commit f672209110ef30b9cdf96048d4e8285d43716b4a.

bretg committed:
Rubicon: adding size 300x200 (#6315)

lksharma committed:
added prebid version support to ixdiag (#6292)

YerkovichM committed:
Extended ID permissions supported by bidder (#6276)

* User id bidder permission scheme

* styling

* prebidServer support

* -

* fix

* prebidServerBidAdapter take eidPermissions directly from userId module

* -

* unit tests

* -

* -

* update

* -

* -

* changed pbjs_bidders to pbjsBidders

* changed pbjsBidders to bidders
ext.prebid.data.eidPermissions to ext.prebid.data.eidpermissions

* rerun circleci

* rerun circleci

* omitting eidPermission entry if 'bidders' is not configured

* fixed 'calling the PBS adapter without any defined userId modules causes an exception'

* proposal

* rerun circleci

* utils implementation

* comment

* revert import validation

Co-authored-by: myerkovich <myerkovich@rubiconproject.com>
Co-authored-by: Marko Yerkovich <marko_yerkovich@MacBook-Pro-Marko.local>
Co-authored-by: Marko Yerkovich <myerkovich@magnite.com>

WlsLogan committed:
new bid adapter: Logain (#6312)

Co-authored-by: Aiholkin <artem.iholkin@smartyads.com>

pcejrowski committed:
Change URL in Tapad Id Submodule (#6318)

li-ran committed:
IndexExchange Bid Adapter: Added support for netID, ID+ and FabrickId userId (#6286)

* added more userID partner support for IX adapter - TDID, LI, NetID, Neustar, Zeotap

* removed liveintent and adsrvr.org [DNA-5040]

* updated neustar id

* fixed fabrickid

* frabrickId instead of id

* added more userID partner support for IX adapter - NetID, Neustar, Zeotap (#1)

* added more userID partner support for IX adapter - TDID, LI, NetID, Neustar, Zeotap

* added tests

* added tests for new userinfo modules

* missed a few checks

* Dna 5040 ix userid support (#2)

* added more userID partner support for IX adapter - TDID, LI, NetID, Neustar, Zeotap

* removed liveintent and adsrvr.org [DNA-5040]

* updated neustar id

* fixed fabrickid

* frabrickId instead of id

* added tests

* added tests for new userinfo modules

* missed a few checks

Co-authored-by: Ran Li <ran.li@indexexchange.com>

* added few missing checks in IX adapter

* Use userIdAsEids instead of userId (#3)

* use eids, revert package-lock
* test update

Co-authored-by: Ran Li <ran.li@indexexchange.com>
Co-authored-by: amykwyang <amyyang90@gmail.com>

jsnellbaker committed:
Prebid 4.27.0 Release

jsnellbaker committed:
increment pre version

matthieularere-msq committed:
mediasquare fix userId (#6321)

slimkrazy committed:
Change the EU consent string parameter name to the company-wide standard (#6320)

robertrmartinez committed:
adding support in pbsAdapter for getFloor (#6273)

wyand-sp committed:
AdHash Bidder Adapter: initial prebid.js integration (#6274)

mmoschovas committed:
PBJS Video Cache Update (#6295)

cciocov committed:
Support bidder aliasing by not using bid.bidder to retrieve configs (which are static anyway) (#6313)

andersonbd1 committed:
Fabrick ID System : updates to _setReferrer and appending urls (#6322)

* fabrickId updates

 - encode all url params
 - dedupe w/out queryString and keep the longest
 - additionally truncate from % if that ends up being the last (or next to last) char after truncation
 - truncate 1k instead of 200
 - don't send functions along in query (only send strings and numbers)

* attempting to fix ie failure - https://app.circleci.com/pipelines/github/prebid/Prebid.js/3945/workflows/4929a5c0-9e57-49e3-a6d8-29691560ca31/jobs/12086

* re-attempting to fix ie failure - https://app.circleci.com/pipelines/github/prebid/Prebid.js/3945/workflows/4929a5c0-9e57-49e3-a6d8-29691560ca31/jobs/12086

* re-attempting to fix ie failure - https://app.circleci.com/pipelines/github/prebid/Prebid.js/3945/workflows/4929a5c0-9e57-49e3-a6d8-29691560ca31/jobs/12086

* re-attempting to fix ie failure - https://app.circleci.com/pipelines/github/prebid/Prebid.js/3945/workflows/4929a5c0-9e57-49e3-a6d8-29691560ca31/jobs/12086

* re-attempting to fix ie failure - https://app.circleci.com/pipelines/github/prebid/Prebid.js/3945/workflows/4929a5c0-9e57-49e3-a6d8-29691560ca31/jobs/12086

* re-attempting to fix ie failure - https://app.circleci.com/pipelines/github/prebid/Prebid.js/3945/workflows/4929a5c0-9e57-49e3-a6d8-29691560ca31/jobs/12086

- attempting to add debugging so I can know what exactly is failing in ie (since I don't have a windows machine)

* found the issue having debugged locally - stubbing my error message is what killed me!

* not the right way to loop through a map

Co-authored-by: Anderson, Ben <Ben.Anderson@team.neustar>

robertrmartinez committed:
Configureable option to delay auction event + fix bug with googletag init (#6344)

mefjush committed:
Adhese Bid Adapter: replace id5 with eid (#6339)

vfedoseev committed:
FID-287: Update Reconciliation RTD delivery id format (#6343)

cciocov committed:
Inskin Bid Adapter: send screen size in the ad call (#6316)

afsheenb committed:
Ozone Bid Adapter: added or updated support for multiple modules (#6324)

* ozone 2.5.0 adapter updates

* ozone 2.5.0 adapter - fixup for prebidCircleCI tests to remove object values on user.eid

Co-authored-by: Afsheen Bigdeli <afsheenb@namecast.net>

nllerandi3lift committed:
Triplelift Bid Adapter: add PubCommon ID support (#6352)

Commit:
Prebid 4.28.0 Release

Commit:
Increment pre version

KazuakiM committed:
GMOSSP Bid Adapter :  add refererInfo from bidderRequest  (#6319)

cesarfd committed:
Axonix Bid Adapter: add new bid adapter (#6341)

* Add Axonix bid adapter

* Fixed tests

ysfbsf committed:
Documentation: Prebid Server and Postbid integration example ( ad server-less ) (#6348)

osazos committed:
Native: add custom data assets capability (#6220)

Use `mediaTypes.native.ext: {}` special object to add/allow custom data. assets.

iprom-adserver committed:
iPROM adapter upload - adapter (#6334)

Co-authored-by: Gašper Žagar <gasper.zagar@iprom.si>

jackhsiehucf committed:
Ucfunnel Bid Adapter: update currency parameter in ad response (#6357)

TheMediaGrid committed:
TheMediaGrid Bid Adapter: fix trouble with alias using (#6363)

robertrmartinez committed:
Rubicon Analytics Adapter: pass along advertiserDomains (#6356)

* Pass along advertiser domains!

* only send up to 10 adomains

msm0504 committed:
support setting coopSync in s2sConfig (#6330)

PWyrembak committed:
TrustX Bid Adapter: Fix alias error (#6373)

Pupis committed:
Adform Bid Adapter: add global targeting to the request query as parameter (#6376)

Co-authored-by: Justas <jpmiltas@gmail.com>

smenzer committed:
Prebid 4.29.0 Release

smenzer committed:
Increment pre version

Commit:
Gamoshi Bid Adapter: update adaptor aliases (#6355)

vladi-mmg committed:
Marsmedia Bid Adapter: add support for viewability, floor price module, COPPA & CCPA (#6337)

* Change publisherId to zoneId
Add gdpr
Add supply chain
Add video media type

* Remove comments

* Fix unit test coverage

* fix request id bug
add vastXml to video response

* Remove bid response default sizes

* Change endpoint url

* Add unit test for vastXml

* Change end point

* Remove trailing-space

* Add onBidWon function

* New adapter - videofy

* Marsmedia & Videofy - Add onTimeout onSetTargeting

* Create sendbeacon function

* - add viewability

* remove unnecessary utils.getWindowTop()

JulieLorin committed:
fix bidderRequest matching for finding a renderer (#6359)

lauramorillo committed:
Seedtag Bid Adapter: add support for inArticle placement (#6369)

* Fix: check mandatory video params

* Simplifying mediaType video existence check

* SQDTAR-42: onWonBid event (#2)

* Adding onBidWon handler.

* Adding nurl to bid.

* Adding nurl field to bid.

* Adding inArticle placement type to seedtag adapter. (#1)

Co-authored-by: Carlos Barreiro Mata <barreymata@gmail.com>

guiann committed:
improve robustness on OnEvent fields (#6384)

adam-browning committed:
oneVideo Bid Adapter: remove adapter adId because of conflict with pbjs core (#6382)

* updated object,str,num validations using pbjs utils

* validation if statements for content object

* validation if statements for content object

* updated contetn object validations using utils

* refractoring clean if statement

* fixing typos

* added todos

* added category string & data object validations

* fixed esling ENDPOINT issue

* updated content obj unit tests

* fixed cat & data validation

* fixed producer as object

* revert .includes() to .indexof() for IE

* reduced content obj params accoriding to ad-server support

* fixed typeOf typo

* fixed episode to Number

* gitignore

* restore gitignore

* removed unsupported params from md file

* reverted package & pagkage-lock

* fixed incorrect episdoe from str to num

* removed bid.adid setting from L97

* removed bid.adId setting

* updated adId test

* updated version to 3.0.6

patmmccann committed:
PBS Bid Adapter: add dchain (demand chain object) to prebid server adapter (#6383)


* Update prebidServerBidAdapter_spec.js

* Update index.js

pm-harshad-mane committed:
Gulp Build: fix to populate modules list when gulp bundle is executed (#6331)

* added support for pubcommon, digitrust, id5id

* added support for IdentityLink

* changed the source for id5

* added unit test cases

* changed source param for identityLink

* need to update modules list when gulp bundle is called

* added a comment

Skylinar committed:
Smartx Bid Adapter: updated out-stream render to support smartPlay 5.2 (#6370)

* Add smartclipBidAdapter

* smartxBidAdapter.js - removed unused variables, removed debug, added window before the outstream related functions

* - made outstream player configurable

* remove wrong named files

* camelcase

* fix

* Out-Stream render update to SmartPlay 5.2

* ESlint fix

* ESlint fix

* ESlint fix

* adjust tests, fixes

* ESlint

Co-authored-by: smartclip AdTechnology <adtech@smartclip.com>
Co-authored-by: Gino Cirlini <adtech@smartclip.tv>

mmoschovas committed:
Fpd 2.0 Update (#6293)

* Update to consolidate applying FPD to both banner and video requests. FPD will be merged using global defined FPD, ad unit FPD, and rubicon bidder param FPD. Validation logic with warning logs added

* Refectored last push to:
1) Correct keywords bug
2) Revise error which looked for FPD in (user/context).ext.data as opposed to (user/context).data
3) General code cleanup

* Consolidated other FPD data logic into new function

* 1. Update to move pbadslot and adserver data into imp[] as opposed to parent.
2. Update to convert keywords passed through RP params to string if array found

* Removed unnecessary conditional

* Changed conditional to check for undefined type

* FPD 2.0 Update
1) The setConfig and setBidderConfig functions support a transition period where they map the original 'fpd' config:
   - fpd.context.ATTR --> ortb2.site.ATTR
   - fpd.context.data.ATTR --> ortb2.site.ext.data
   - fpd.user.ATTR --> ortb2.user.ATTR
   - fpd.user.data.ATTR --> ortb2.user.ext.data
2) gptPreAuction:
   a) move adunit.fpd to adunit.ortb2
   b) adUnit.ortb2Imp.ext.data.adserver.{name, adSlot}
   c) pbAdSlot moves to AdUnit.ortb2Imp.ext.data.pbAdSlot

3) pbsBidAdapter
   a) merge the new ortb2 and AdUnit.ortb2Imp.ext objects into the OpenRTB JSON.
   b) therefore imp[].ext.context.data.pbadslot is now changed to imp[].ext.data.pbadslot (no context)
   c) read adUnit.ortb2Imp.ext.data.adserver from the new location. Output location is moved to imp[].ext.data.adserver (no context)

* FPD 2.0 Update
Update to adrelevantis adapter to look at config.ortb2 instead of config.fpd

* FPD 2.0 Update
Update to amx adapter to look at config.ortb2 instead of config.fpd

* FPD 2.0 Update
Update to avocet adapter to look at config.ortb2 instead of config.fpd

* FPD 2.0 Update
Update to criteo adapter to look at config.ortb2 instead of config.fpd

* Update to correct imp fpd structure

* Update to s2s adapter to coincide with imp fpd alteration

* Update to consolidate several lines of duplicate code into one location

* Slight modification for ortb2Imp to use ortb2Imp.ext as opposed to ortb2Imp.ext.data

* FPD 2.0 Update
Update to grid adapter to look at config.ortb2 instead of config.fpd

* FPD 2.0 Update
Update to inmar adapter to look at config.ortb2 instead of config.fpd

* FPD 2.0 Update
Update to luponmedia adapter to look at config.ortb2 instead of config.fpd

* FPD 2.0 Update
Update to smaato adapter to look at config.ortb2 instead of config.fpd

* FPD 2.0 Update
Update to triplelift adapter to look at config.ortb2 instead of config.fpd

* Update to gptPreAuction to move over to imp level ortb2

* FPD 2.0 Update
Update to triplelift adapter to look at config.ortb2 instead of config.fpd

* FPD 2.0 Update

* FPD 2.0 Update
Update to jwplayerRtd adapter to look at config.ortb2 instead of config.fpd

* FPD 2.0 Update
Update to admixer adapter to look at config.ortb2 instead of config.fpd

* FPD 2.0 Update
Update to rubicon adapter to look at config.ortb2 instead of config.fpd

* Update to fix keyword bug

* Added backwards compatibility functions for FPD both global/bidder and adunit level data

* Update to utilize new backward functionality for fpd 2.0

* Removed extra new line

* Update to include new backward functionality for FPD 2.0 data

* Update to utilize new backward functionality to pass FPD 2.0 data

* Update to utilize backward functionality to pass FPD 2.0 data

* Update to utilize backward functionality to pass FPD 2.0 data

* Update to utilize backward functionality to pass FPD 2.0 data

* Update to utilize backward functionality to pass FPD 2.0 data

* Update to utilize backward functionality to pass FPD 2.0 data

* Fixed typo in fpd config object location

* Uodate to utilize backward functionality to pass FPD 2.0 data

* Update to change all ortb2Imp.ext.data.adserver.adSlot references to ortb2Imp.ext.data.adserver.adslot - all lowercase. Corresponding adapter and unit tests to adhere to these changes

* Fixed typo

* Fixed typo

* FPD 2.0 update to rubicon adapter to pass iab values

* Updates:
1) Change function name
2) addAdUnits always pass array
3) Remove unecessary comment
4) Bug fix for ortb2.user.data to be filtered on legacy fpd conversion

Commit:
PBS Bid Adapter: fix s2s alias collision with built-in adapter aliasing (#6379)

* fixed overwriting of aliases for s2s

* made change

* added fix

TurpIF committed:
Clean side-effect when checking that local storage is enabled (#6323)

karimMourra committed:
updates docs and demo for fpd changes (#6302)

Co-authored-by: karimJWP <karimJWP@github.com>

pm-azhar-mulla committed:
Changed net revenue to True (#6387)

osazos committed:
adagioBidAdapter: add Native support (#6368)

rtuschkany committed:
EIDS Support Update (#6394)

TheMediaGrid committed:
TheMediaGridNM Bid Adapter: fix trouble with alias (#6371)

* Added TheMediaGridNM Bid Adapter

* Updated required params for TheMediaGridNM Bid Adapter

* Update TheMediGridNM Bid Adapter

* Fix tests for TheMediaGridNM Bid Adapter

* Fixes after review for TheMediaGridNM Bid Adapter

* Add support of multi-format in TheMediaGrid Bid Adapter

* Update sync url for grid and gridNM Bid Adapters

* TheMediaGrid Bid Adapter: added keywords adUnit parameter

* Update TheMediaGrid Bid Adapter to support keywords from config

* Implement new request format for TheMediaGrid Bid Adapter

* Fix jwpseg params for TheMediaGrid Bid Adapter

* Update unit tests for The Media Grid Bid Adapter

* Fix typo in TheMediaGrid Bid Adapter

* Added test for jwTargeting in TheMediaGrid Bid Adapter

* The new request format was made by default in TheMediaGrid Bid Adapter

* Update userId format in ad request for TheMediaGrid Bid Adapter

* Added bidFloor parameter for TheMediaGrid Bid Adapter

* Fix for review TheMediaGrid Bid Adapter

* Support floorModule in TheMediaGrid Bid Adapter

* Fix empty bidfloor for TheMediaGrid Bid Adapter

* Some change to restart autotests

* Fix userIds format for TheMediaGrid Bid Adapter

* Remove digitrust userId from TheMediaGrid Bid Adapter

* Protocols was added in video section in ad request for TheMediaGrid Bid Adapter

* TheMediaGrid: fix trouble with alias using

* TheMediaGridNM: fix trouble with alias

robertrmartinez committed:
Grab sourceAgnostic IDs first, then fallback to regular IDs (#6400)

dreischer committed:
Permutive - add AC support for TrustX (#6393)

evanmsmrtb committed:
SmartRTB Bid Adapter: add alias and update valid opts (#6365)

* Add alias, update valid opts

* Update bidder tests

robertrmartinez committed:
Prebid 4.30.0 Release

robertrmartinez committed:
Increment pre version

jongo593 committed:
Sonobi Bid Adapter: send eids in bid request. (#6364)

* unwrapping id5id uid. Added new eid param for user id modules

* set userid to new variable

* fixed spelling mistake in unit test

Co-authored-by: Scott Menzer <smenzer@gmail.com>

* copying userid object so the referenced object does not get updated.

* using deepClone instead of spreading the top userId object

Co-authored-by: Scott Menzer <smenzer@gmail.com>

thuyhq committed:
Apacdex Bid Adapter: userId module support, show demo ads in debug mode & other maintenance (#6378)

* Upgrade and maintenance apacdexBidAdapter

* fix error and add unit test

Commit:
Novatiq ID System: add snowflake userId submodule (#6350)

* Novatiq snowflake userId submodule

Novatiq snowflake userId submodule initial release

* change request updates

added novatiq info /modules/userId/userId.md
added novatiq info /modules/userId/eids.md
added novatiq eids /modules/userId/eids.js
added novatiq module in /modules/.submodules.json
removed unnecessary value from getId response

* Update novatiqIdSystem_spec.js

removed unnecessary srcid value

* Update novatiqIdSystem.md

Novatiq ID System: updated novatiq snowflake ID description

lasloche committed:
Rise Bid Adapter: add session_id & is_wrapper params to adapter (#6407)

* add new params to rise adapter

* add unit tests for isWrapper and sessionId adapter params

lasloche committed:
IronSource Bid Adapter: add session_id & is_wrapper params to adapter (#6408)

* add new params

* add unit tests for isWrapper and sessionId adapter params

nickjacob committed:
AMX Bid Adapter: add or update general adapter support and code refactoring (#6403)

* AMX Bid adapter improvements

* fix eslint issues (breaking CI)

hybrid-ai committed:
VOX Bid adapter, Hybrid Bid adapter: fix global var name to avoid conflicts with astraOne adapter. (#6416)

Co-authored-by: Petrov Denis <d.petrov@hybrid.ai>

Commit:
LiveIntent Id System: fix for parsing response twice  (#6418)

* Don't parse response twice in LiveIntent Id submodule

* Update the liveintent module test to have the 204 response

aleksatr committed:
PBJS Core: use mediaType renderer when backupOnly and no bid.renderer (#6419)

* use mediaType renderer when backupOnly and no bid.renderer

* check if necessary renderer properties are defined

susyt committed:
GumGum Bid Adapter: pass bidfloor currency in bidrequest (#6391)

* adds support for zone and pubId params

* adds support for iriscat field

* sets mediatype depending on product id

* Update doc for mediaType needed for video products

* makes slot and invideo products avail for pubId

* updates gumgum doc

* lint

* adds missing comma in gumgum doc

* adds currency in ad request, adds unit test

* readd the previous irisid changes

* remove the only in testing

ChrisHuie committed:
Documentation: fixed a typo and sentence structure (#6421)

dliebner committed:
Bid Glass Bid Adapter: pass options in bid request (#6424)

* Added bidglass adapter + test

* PR Review Updates:

- Added formal params to getUserSyncs function definition
- getUserSyncs now always returns an array
- Improved unit test coverage

* PR Review Updates:

- Removed unused methods: getUserSyncs, onTimeout, onBidWon,
onSetTargeting
- Removed getUserSyncs unit test
- Removed "dead code"
- Removed some unnecessary comments
- Fixed usage of parseInt

* Bid Glass Bid Adapter: pass options in bid request

npeceniak committed:
Spotx Bid Adapter: add publisher support for cache.ignoreBidderCacheKey (#6413)

* Support ignoreBidderCacheKey in spotxBidAdapter

* Update spotxBidAdapter_spec.js

* Update spotxBidAdapter_spec.js

Co-authored-by: Nick Peceniak <npeceniak@spotx.tv>

BizzClick committed:
update prebid adapter. Add at, ccpa, gdpr and coppa support (#6405)

pro-nsk committed:
Qwarry Bid Adapter: add referer detection  (#6412)

* qwarry bid adapter

* formatting fixes

* fix tests for qwarry

* qwarry bid adapter

* add header for qwarry bid adapter

* bid requests fix

* fix tests

* response fix

* fix tests for Qwarry bid adapter

* add pos parameter to qwarry bid adapter

* qwarryBidAdapter onBidWon hotfix

* Change bidder endpoint url for Qwarry adapter

* add referer JS detection

* use bidderRequest.refererInfo

* fix tests

Co-authored-by: Artem Kostritsa <akostritsa@akostritsa.com>
Co-authored-by: Alexander Kascheev <akascheev@asteriosoft.com>

SmartyAdsSSP committed:
Smartyads Bid Adapter: add coppa field from config (#6402)

* update adapter. Add coppa field from config

* move stubs and restores for coppa tests

mmoschovas committed:
PBS Bid Adapter: Fpd2.0 bug fix for first party data issue (#6428)

* Bug fix for PBS data from FPD2.0 update: Merging request.site and request.user with site and user object in FPD.

Commit:
Prebid 4.31.0 Release

Commit:
Increment pre version

Junus committed:
a4g Bid Adapter: delete adid and use crid if it exists (#6409)

* Deleted adid

* set crid if it's exist and added unit tests

ericrenebrown committed:
Mediawallah ID System: add openlink userId submodule (#5921)

* My first commit

* Removed unnecessary await operation

* Bug fixes and compliance fixes

* Fixing some formatting and naming

* Updating code based on automated feedback.

* Parking refactoring change for team review

* update mwOpenLink module

* remove .git 2 folder

* Trying to force a change

* update the PR comments

* applying the changes

* update submodules.json and userId.md

* fix typo of module names

* update module decode function and test code

* update test codes

* apply the suggestions from Prebid

* fix count of modules.

Co-authored-by: Eric Brown <eric@mediawallah.com>
Co-authored-by: hanna <hanna.panova190119@gmail.com>
Co-authored-by: hannapanova190119 <71532550+hannapanova190119@users.noreply.github.com>

Skylinar committed:
Documentation: Adjust desired bitrate examples smartx adapter (#6438)

jbartek25 committed:
Remove adId (autogenerated by Prebid) (#6441)

richaudience committed:
RichAudience Bid Adapter: add render video in banner (#6392)

SKOCHERI committed:
Integration Example: ID import library example (#6434)

* ID Import Library example

* Fixing review comments

Co-authored-by: skocheri <skocheri@rubiconproject.com>

cciocov committed:
Mass Module: add module to support MASS protocol (#6332)

* Initial implementation for the MASS module

* Updated namespace and CDN location

* Updated the data object passed to MASS bootloader

* Fix linting issues

* Added unit tests

* Added a README for the MASS module.

* Allow MASS be disabled via Prebid configuration

* Only check deal ID for matching MASS bids

* Updated docs

* Update how we test for MASS bids

* Thighten the test for MASS bids

* Fix linting issues

* Change deal ID prefix and add option to specify the bootloader location. Updates to docs.

* Updated tests with the new META_MASS deal ID prefix

* Update comment in modules/mass.js

Co-authored-by: Scott Menzer <smenzer@gmail.com>

* Additional information about the module

* More specific description of external resource

* Identify MASS bids by looking for a 'mass' flag in bid meta or testing deal IDs against a publisher defined pattern

* Updated MASS module tests

* Bug fixing, added integration example and increased test coverage

* Fix integration example and add notice

* Updated example page

* Updated bootloaderUrl param name to renderUrl and removed its default value. Must be specfied in module config now.

* Updated integration example for MASS

* Update mass.md

Updated disclaimer and synced with docs

Co-authored-by: Scott Menzer <smenzer@gmail.com>
Co-authored-by: massadmin <58946787+massadmin@users.noreply.github.com>

Commit:
Axonix bid adapter: set both connectiontype & effectivetype in the request (#6439)

* Set both connectiontype and effectivetype in the request object

* Version bump

* Fixes for undefined var

JeremiePat committed:
Sublime Bid Adapter: v0.7.1 & add extra information in tracking pixels (#6442)

* add prebid version of adapter

* Feature/update sublime adapter (#21)

* Update sublimeBidAdapter to 0.5.1
* Add tests for private functions
* Remove window.sublime
* Update pixel name for bid event

* Remove pixels on non-event and add onBidWon (#22)

* add prebid version of adapter
* Feature/update sublime adapter (#21)
* Update sublimeBidAdapter to 0.5.1
* Add tests for private functions
* Remove window.sublime
* Update pixel name for bid event
* Remove pixels on non-event and add onBidWon
* Incremente version of sublimeBidAdapter
* Renamed pixel for timeout and introduce gvlid
* Remove unnecessary params for sendEvent

Co-Authored-By: fgcloutier <fg.cloutier@sublimeskinz.com>
Co-authored-by: Gaby <gaby.hourlier@sublimeskinz.com>
Co-authored-by: fgcloutier <fg.cloutier@sublimeskinz.com>

* Remove trailing-space

* Fix version in tests

* sublimeAdapter: Improve pixels data

* sublimeAdapter: Update tests

* sublimeAdapter: Rename data

Co-authored-by: Gaby <gaby.hourlier@sublimeskinz.com>
Co-authored-by: Léo <51166933+SublimeLeo@users.noreply.github.com>
Co-authored-by: fgcloutier <fg.cloutier@sublimeskinz.com>
Co-authored-by: Léo GRAND <leo.grand@sublimeskinz.com>

IOTiagoFaria committed:
Interactive Offers Bid Adapter: add new bid adapter (#6399)

* Interactive Offers bid adapter

* Interactive Offers bid adapter

* Interactive Offers bid adapter

* Interactive Offers bid adapter

* Interactive Offers bid adapter

* Interactive Offers bid adapter

* Interactive Offers bid adapter

* Interactive Offers bid adapter

* manually kicking off tests

Co-authored-by: Chris Huie <phoenixtechnerd@gmail.com>

guiann committed:
AdYouLike Bid Adapter: replace shorthand "image" native config (#6401)

* replace shrorthand "image" native type

* merge publisher native config with image type config

* add unit tests on native image config type

nickjacob committed:
AMX Bid Adapter: Video Bugfix (#6435)

bjorn-lw committed:
LiveWrapper Bid Adapter: use userIdAsEids to handle all user ids (#6445)

* Livewrapped bid and analytics adapter

* Fixed some tests for browser compatibility

* Fixed some tests for browser compatibility

* Changed analytics adapter code name

* Fix double quote in debug message

* modified how gdpr is being passed

* Added support for Publisher Common ID Module

* Corrections for ttr in analytics

* ANalytics updates

* Auction start time stamp changed

* Detect recovered ad blocked requests
Make it possible to pass dynamic parameters to adapter

* Collect info on ad units receiving any valid bid

* Support for ID5
Pass metadata from adapter

* Typo in test + eids on wrong level

* Fix for Prebid 3.0

* Fix get referer

* http -> https in tests

* Native support

* Read sizes from mediatype.banner

* Revert accidental commit

* Support native data collection + minor refactorings

* Set analytics endpoint

* Support for app parameters

* Fix issue where adunits with bids were not counted on reload

* Send debug info from adapter to external debugger

* SChain support

* Send GDPR data in analytics request

* video support

Video support

* Report back floor via analytic

* Send auction id and adunit/bidder connection id

* Criteo id support

* Updated example

* livewrapped Analytics Adapter info file

* Livewrapped gvlid

* Handle all User Ids

* kick off tests manually

Co-authored-by: Chris Huie <phoenixtechnerd@gmail.com>

invibes committed:
Invibes: Read legitimate interests; pubCommonId & pubProvidedId integration (#6420)

patmmccann committed:
Update dfpAdServerVideo.js

patmmccann committed:
revert accidental commit

haibau committed:
AdTrue Bid Adapter: getUserSyncs logic update (#6338)

* add AdTrue bid adapter

* add AdTrue Bid Adapter unit tests

* add AdTrue Bid Adapter unit tests

* add AdTrue bid adapter

* add AdTrue Bid Adapter unit tests

* add AdTrue Bid Adapter unit tests

* add AdTrue bid adapter

* adtrue usersync udpate

* adtrue usersync udpate

* #adtrue bidder usersync udpate

* #Refactoring code & add unit test

* Update package.json

* Update package-lock.json

* 1. Merge branch 'master' of https://github.com/prebid/Prebid.js
2.AdTrue Bid Adapter Refactoring code

* 1. Merge branch 'master' of https://github.com/prebid/Prebid.js
2.AdTrue Bid Adapter Refactoring code

* removing a package-lock.json from PR

* - removing a package-lock.json from PR
- AdTrue bidder: refactoring code & add unit test

* - removing a package-lock.json from PR
- AdTrue bidder: refactoring code & add unit test

* - removing a package-lock.json from PR
- AdTrue bidder: refactoring code & add unit test

* update package-lock.json from upstream

punkiller committed:
IX Bid Adapter: add priceFloors support and an integration example (#6390)

* Adding priceFloors Support to IX bid adapter and some refactoring

* refactoring the getfloor logic, and removing dead code

* newline at end of integration example

* spacing issues lint fix

* try catch around getFloor, in case the priceFloors module throws an exception

* minor rename allU -> allu

* falsey values fix, log prefix added

patmmccann committed:
PBS adapter: remove condition on gdpr consentstring which isnt required by the tcf2 spec when gdprapplies is false (#6429)

* Update index.js

* Update prebidServerBidAdapter_spec.js

* Update prebidServerBidAdapter_spec.js

* Update prebidServerBidAdapter_spec.js

smenzer committed:
if the stored value was an object, then it could end up with a space before the {, causing the stored value not to be json parsed simply because of a space in the first character (#6467)

BernhardValenti committed:
Kargo Bid Adapter: add gdpr support (#6447)

dliebner committed:
Bid Glass Adapter: Merge externally set targeting params (#6426)

* Added bidglass adapter + test

* PR Review Updates:

- Added formal params to getUserSyncs function definition
- getUserSyncs now always returns an array
- Improved unit test coverage

* PR Review Updates:

- Removed unused methods: getUserSyncs, onTimeout, onBidWon,
onSetTargeting
- Removed getUserSyncs unit test
- Removed "dead code"
- Removed some unnecessary comments
- Fixed usage of parseInt

* Bid Glass Bid Adapter: pass options in bid request

* Merge externally set targeting params

* Updates to address gulp errors

* Get `bidglass` reference from window

pixelgroup-israel committed:
Optimon Analytics Adapter: add new analytics adapter (#6333)

* Publish the Optimon platform's analytics adapter for prebid

Added js and md files for the analytics adapter.

* Fix wrong content in MD file

* Fix wrong content in MD file

* Created unit testing to Optimon Analytics Adapter

* Created unit testing to Optimon Analytics Adapter

* Created unit testing to Optimon Analytics Adapter

* Created unit testing to Optimon Analytics Adapter

* Fixes ESlint styling

* Removing DEF const

* Created unit testing to Optimon Analytics Adapter

* Created unit testing to Optimon Analytics Adapter

mkendall07 committed:
4.32.0 release

mkendall07 committed:
4.33.0-pre

SKOCHERI committed:
UID 2.0 Userid submodule (#6443)

* UID 2.0 User id submodule

* UID 2.0 User id submodule

* UID 2.0 User id submodule

* UID 2.0 User id submodule

* Resolving merge conflicts and review comments

* Updating documentation

* Renaming module

* Fixing review comments

* Fixing review comments

* Fixing review comments

* Fixing review comments

* Fixing review comments

* Updating source uid2.com to uidapi.com

Co-authored-by: skocheri <skocheri@rubiconproject.com>

h12media committed:
H12media Bid Adapter: added new optional params, fixes, and various integration support (#6436)

* Change module H12 Media

* Change module H12 Media

* Change module H12 Media

* Change module H12 Media

* Update module H12 Media

* Update module H12 Media

* Update module H12 Media

* Update module H12 Media

* Update module H12 Media

* Update module H12 Media

guiann committed:
AdYouLike Bid Adapter: add video capabilities (#6398)

* read and send getFloor value

* improve robustness on OnEvent fields

* add video mediatype to adyoulike  adapter

* improve robustness on OnEvent fields

* handle instream cases

* minor format fix

* move the atob call to apply it on every Vast response

* update adyoulike md file with video type

* update adyoulike bidder doc

* fix merge error on 'bid' var name

* update adyoulike bidder doc

ChrisHuie committed:
remove deprecated creative rendering example

GLStephen committed:
Update Recommendation to install with 'npm ci' (#5697)

With the recent upgrade of npm we should be using the 'npm ci' option install packages. This provides a more reliable deployment that does not update packages on install. It will reduce check-ins of improper package-lock.json and provide greater assurance that a stable node_modules is deployed.

kenan-gillet committed:
Openx Bid Adapter: add floor support for Video (#6460)

* fix confusing comment

* refactor(openxBidAdapter): extract enrichQueryWithFloors

* PBID-1194: add floor for Video

mmoschovas committed:
Update to rubicon analytics adapter to pass along pbsBidId if present on the bidObject (#6461)

smenzer committed:
differentiate the IDs provided by ID5 when the user is opted out vs the user is in the control group of A/B testing (#6471)

osazos committed:
AdagioBidAdapter: add capability to pass ext-data from localStorage in bidRequest (#6415)

ybootin committed:
Seedtag Bid Adapter: add connection type to bid request (#6457)

* add connectionType to the bidRequest

* remove a console.log

* put connectionType in the payload

* more secure connectionType access

* add a unit test for connectionType

* fix  connectionType typo

addefend committed:
AdDefend Bid Adapter: new bid adapter (#6450)

* addefeend bid adapter implementation

* fixed parameter documentation

Co-authored-by: Mitschi <mitschi@addefend.com>

susyt committed:
GumGum Bid Adapter: adds meta field to bidresponse (#6478)

* adds meta field to bidresponse

* adds meta mediatype and advertiserdomain default

ChrisHuie committed:
example integration file renamed to hello_world (#6475)

patmmccann committed:
Incorrect atypes on various eids (#6477)

* Update eids.js

* Update eids_spec.js

* Update eids.js

* Update pubmaticBidAdapter_spec.js

* Update eids.js

* Update eids_spec.js

* Update conversantBidAdapter_spec.js

* Update rubiconBidAdapter_spec.js

* Update conversantBidAdapter_spec.js

* Delete test/spec/adapters directory

* Update userId_spec.js

lksharma committed:
added support for roundel alias (#6473)

sourbh-doceree committed:
Doceree Bid Adapter: new bid adapter (#6437)

* doceree adapter

* add fix for impression register

* fix linting issues

* fix build issue

* remove extra spaces

* fixing - Implicit operand conversion

* fixed ci/cd lint issue

* update test case

* support for advertiserdomains and updated test case

* remove space to rerun test cases

* remove the client side creative creation

kenan-gillet committed:
OpenX Bidder Adapter: Add user ID sub-modules (#6490)

el-chuck committed:
Smaato: FPD Revision (#6459)

Co-authored-by: Bernhard Pickenbrock <bernhard.pickenbrock@smaato.com>

patmmccann committed:
Conversant adapter: add adomain, remove digitrust (#6495)

* Update eids.js

* Update eids_spec.js

* Update eids.js

* Update pubmaticBidAdapter_spec.js

* Update eids.js

* Update eids_spec.js

* Update conversantBidAdapter_spec.js

* Update rubiconBidAdapter_spec.js

* Update conversantBidAdapter_spec.js

* Delete test/spec/adapters directory

* Update userId_spec.js

* Update conversantBidAdapter.js

* Update conversantBidAdapter_spec.js

* Update conversantBidAdapter_spec.js

* Update conversantBidAdapter_spec.js

onlsol committed:
Rads Bid Adapter: add GDPR support & user sync support (#6455)

vincentproxistore committed:
Proxistore Bid Adapter: add cookieless url endpoint & use floor module (#6427)

* use floor module

* call cookieless endpoint when necessary

* test endpoint url

* change url endpoint

* delete console log

* fix tests

* add language to url

* use ortb interface

* unit test

* update test unit

* create proxistore module

* add unit tests and documentation

* delete modules

* delete module

* add proxistore rtd submodule

* delete proxistore module

* spacing

* change url

guiann committed:
AdYoulike Bid Adapter: Add an "Insertion" tracking for Native mediatype (#6481)

* add insertion event

* add missing campaign ID parameter

* update unit test with new tracking checked

onlsol committed:
Dspx Bid Adapter : add user sync support (#6456)

* Add sync support for dspx adapter

* Dspx Bid Adapter : add user sync support

Co-authored-by: Alexander <avj83@list.ru>

mmoschovas committed:
Multibid Module: add new module to handle multiple bids from single bidder & update rubicon adapter (#6404)

* Multibid module - create new module
- Expands the number of key value pairs going to the ad server in the normal Prebid way by establishing the concept of a "dynamic alias"

First commit

* Continued updates from 1st commit

* Adding logWarn for filtered bids

* Update to include passing multibid configuration to PBS requests

* Update to rubicon bid adapter to pass query param rp_maxbids value taken from bidderRequest.bidLimit

* Update to config to look for camelcase property names according to spec. These convert to all lowercase when passed to PBS endpoint

* Adjust RP adapter to always include maxbids value - default is 1

* Added support for bidders array in multibid config

* Fixed floor comparison to be <= bid cpm as oppossed to just < bid cpm. Updated md file to fix camelCase tpyo

* Update to include originalBidderRequest in video call to prebid cache

* Update to ignore adpod bids from multibid and allow them to return as normal bids

SKOCHERI committed:
Adding uid2 to submodules.json (#6508)

abijr committed:
NextRoll ID System: add new ID module (#6396)

* Add Nextroll ID Module

* Add nextroll to eids

* Make configuration value names consistent with Adapter Module

* Use parnerId instead of sellerId

* Add nextroll to userId and eids md files

* Remove storage configuration

* Rename nextroll -> nextrollId

* Add nextrollId to common ID specs

artemiokost committed:
Qwarry Bid Adapter: add GDPR and consent string handling (#6489)

* qwarry bid adapter

* formatting fixes

* fix tests for qwarry

* qwarry bid adapter

* add header for qwarry bid adapter

* bid requests fix

* fix tests

* response fix

* fix tests for Qwarry bid adapter

* add pos parameter to qwarry bid adapter

* qwarryBidAdapter onBidWon hotfix

* Change bidder endpoint url for Qwarry adapter

* add referer JS detection

* use bidderRequest.refererInfo

* fix tests

* GDPR consent string support

* NPE fix

Co-authored-by: Artem Kostritsa <akostritsa@akostritsa.com>
Co-authored-by: Alexander Kascheev <akascheev@asteriosoft.com>
Co-authored-by: pro-nsk <32703851+pro-nsk@users.noreply.github.com>

Commit:
Zemanta Bid Adapter: add support for new params & consent strings to usersync URL (#6468)

* add gvl id to spec

* add support for bcat and badv params

* add consent strings to usersync url

* add bcat and badv params to doc

Commit:
Automatad Bid Adapter: Add meta.advertiserDomains to bid response (#6509)


* added bid meta with advertiserDomains

Commit:
Adhese Bid Adapter: add support for caching video content (#6501)

* adpod category support test

* Revert "adpod category support test"

This reverts commit 70a3cf2ad5db94757addd9e08c3a083caca282d0.

* adpod category support test

* Revert "adpod category support test"

This reverts commit 70a3cf2ad5db94757addd9e08c3a083caca282d0.

* Adhese Bid Adapter: cache video content

Co-authored-by: Tim Sturtewagen <tim@adhese.com>
Co-authored-by: Mateusz <mateusz@adhese.eu>

jsnellbaker committed:
update apacdex unit test to disable debug mode (#6511)

bretg committed:
Telaria: not setting adid (#6507)

jsnellbaker committed:
Prebid 4.33.0 Release

jsnellbaker committed:
increment pre version

bretg committed:
rubicon: removing maxduration as a required bidder parameter (#6513)

Commit:
Zemanta adapter: add advertiserDomains (#6517)

lm-abhijit committed:
Lemma Bid Adapter: accepting the floor to use the getFloor function (#6497)

* lemmaBidAdapter.js

Added lemma bid adapter file

* lemmaBidAdapter.md

Added lemma bid adapter md file

* lemmaBidAdapter_spec.js

Added lemma bid adapter test spec file

* Update lemmaBidAdapter.js

Fixed automated code review alert comparison between inconvertible types

* Update lemmaBidAdapter.js

Fixed review changes

* Update lemmaBidAdapter.md

Correct parameter value.

* Update lemmaBidAdapter.js

Lemma Bid Adapter - v3.0 compliance

* Update lemmaBidAdapter_spec.js

Lemma Bid Adapter - v3.0 compliance

* Update lemmaBidAdapter.md

Lemma Bid Adapter - v3.0 compliance

* Update lemmaBidAdapter.js

Added user sync support into bid adapter.

* updated include modules file extension.

updated include modules js file extension.

* Update lemmaBidAdapter_spec.js

Added unit test for user sync feature.

* Update lemmaBidAdapter.js

Fixed format error.

* Update lemmaBidAdapter_spec.js

Fixed format error and typo error.

* Set mediaType key value into bid object

Set mediaType key value into the bid object.

* Update lemmaBidAdapter.js

remove duplicate function

* Update lemmaBidAdapter.js

Remove non supported code.

* Update lemmaBidAdapter_spec.js

Remove GDPR test cases.

* Update lemmaBidAdapter.js

Made changes for accepting the floor to use the getFloor function

* Update lemmaBidAdapter.js

correct undefined keyword name.

* Update lemmaBidAdapter_spec.js

Added test coverage floor value

* Update lemmaBidAdapter.js

Remove trailing spaces on lines 379 and 381.

* Update lemmaBidAdapter_spec.js

Added getFloor function test case changes, Please review it.

* Update lemmaBidAdapter_spec.js

* Update lemmaBidAdapter.js

* Update lemmaBidAdapter.js

Fixed lint issue.

* Update lemmaBidAdapter_spec.js

Fixed test cases.

* Update lemmaBidAdapter_spec.js

Made suggested changes. Please review it.

Co-authored-by: Abhijit Mane <abhipucsd.123@gmail.com>

matthieularere-msq committed:
Mediasquare Bid Adapter: fix getUserSyncs issue with empty bids + add metrics to onBidWon Event (#6480)

* Mediasquare bidder: fix getUserSyncs issue with empty bids + add metrics to onBidWon Event

* Mediasquare bidder: fix getUserSyncs issue with empty bids + add metrics to onBidWon Event

* removing status as it does not seem populated when called

* add tests

abijr committed:
Update nextroll ID variable name to match published ID module (#6519)

SKOCHERI committed:
Merkle User ID Module: updates to user id submodule (#6503)

ckbo3hrk committed:
AdKernel Bid/Analytics Adapters: user privacy related changes (#6488)

rajcspsg committed:
SynacorMedia: remove adId from the bid response (#6520)

bretg committed:
Rubicon: making doc data types consistent (#6526)

rajcspsg committed:
Synacormedia Bid Adapter: add meta.advertiserDomains (#6527)

jimdigriz committed:
Adloox Analytics Adapter: add new analytics adapter (#6308)

* gulp: fix supplying list of browsers to test against

The following now works: gulp test --browserstack --nolint --nolintfix --browsers=bs_ie_11_windows_10 --file 'test/spec/modules/adloox{AnalyticsAdapter,AdServerVideo,RtdProvider}_spec.js'

* instreamTracking: unit test tidy

From @robertrmartinez in https://github.com/prebid/Prebid.js/pull/6308#issuecomment-810537538

* adloaderStub: expose stub for other unit tests to use

From @robertrmartinez in https://github.com/prebid/Prebid.js/pull/6308#issuecomment-810537538

* Adloox analytic module

lauramorillo committed:
Seedtag adapter: Fixing bug preventing to receive the right params onTimeout. (#6525)

Giudici-a committed:
adot bid adapter: add publisher path from bidder config to endpoint url (#6476)

Galphimbl committed:
Admixer ID System: add userId submodule  (#6238)

* Migrating to Prebid 1.0

* Migrating to Prebid 1.0

* Fix spec

* add gdpr and usp

* remove changes in gdpr_hello_world.html

* Update gdpr_hello_world.html

add spaces

* add user syncs

* remove comments

* tests

* admixer id system

* admixer id system

* admixer id system eids.md userId.md

* admixer id system .submodules.json

* admixer id system

Co-authored-by: atkachov <atkachov91@admixer.ua>

jeremiegirault committed:
PBJS Core: call custom render func after loadscript if provided (#6422)

pnhegde committed:
Pubxai Analytics Adapter: bug fixes and code revamp (#6474)

* Updated PubxAiAnalytics adapter - Bug fixes and Code restructuring

* Updated endpoint URLs

* Changed array.includes to array.indexOf to fix IE issue

* Code cleanup and changes as suggested.

* Updated browser testing order and edge browser token

patmmccann committed:
PBJS Core: canBidderRegisterSync ignoring iframe sync disabled by default (#6535)

* Update eids.js

* Update eids_spec.js

* Update eids.js

* Update pubmaticBidAdapter_spec.js

* Update eids.js

* Update eids_spec.js

* Update conversantBidAdapter_spec.js

* Update rubiconBidAdapter_spec.js

* Update conversantBidAdapter_spec.js

* Delete test/spec/adapters directory

* Update userId_spec.js

* Update userSync.js

* Update userSync_spec.js

mikael-lundin committed:
Added automatic tzo and targetId to adserver request. (#6534)

prebidtappx committed:
Tappx Bid Adapter: minor updates and universal id added (#6504)

* UPDATE: add initial UID

* UPDATE: UID change user ext

* UPDATE: UID clean logs

* UPDATE: add host info

* UPDATE: tappx bid adapter universal id

* UPDATE: fix bidder param

* UPDATE: tappxBidAdapter tests

* tappxBidAdapter - fix spacing

* tappxBidAdapter: add test  user eids array

* tappxBidAdapter: update eids array

* FIX: package-lock.json

Co-authored-by: marc_tappx <marc@tappx.com>

thomasdseao committed:
Impactify Bid Adapter: add new bid adapter (#6518)

* Add impactify adapter with MD file

* Add impactify adapter

Commit:
Prebid 4.34.0 Release

Commit:
Increment pre version

sander-adhese commented:
Looks like the last 2 official commits are not building? You might want to keep an eye on that ;-)
Otherwise ok!

deivydas committed:
Prebid server adapter: add config for openx hosting (#6530)

ym-atsymuk committed:
Yieldmo adapter: add meta data to bids (#6550)

Skylinar committed:
Smartx Bid Adapter: Add meta.advertiserDomains support (#6547)

adam-browning committed:
Onevideo / Adap.tv Adapter: updated example configuration (#6546)

cciocov committed:
Mass Deal Rendering Module: support multiple custom configs for dealId and rendering (#6500)

asurovenko-zeta committed:
ZetaSsp Bid Adapter: add new bid adapter (#6432)

mikael-lundin committed:
Adnuntius Bid Adapter: Fix for bid too low. (#6557)

* Added automatic tzo and targetId to adserver request.

* Fixing issues with bid price being too low.

* Fixing issues with bid price being too low.

readpeaktuomo committed:
ReadPeak Bid Adapter: fix api issues, add gdpr consent, & getfloor module support (#6548)

ignat-one committed:
BetweenBidAdatper: added sharedid support (#6531)

nyakove committed:
adWMG Bid Adapter: update endpoints for cookie sync (#6544)

* Support floorCPM parameter, fix some minor bugs

* fix space-in-parens circleci error

* example fix

* clean usersync URL

* spaces

* spaces

* add new unit tests, compatibility with IE11

* remove logInfo

* Check for floorCPM value

* Check params before sending

* New endpoints

* code format

* new endpoint for cookie sync

* update tests

Co-authored-by: Mikhail Dykun <m.dykun@wmgroup.us>

ym-dlabuzov committed:
Yieldmo Bid Adapter: add support for the floors module (#6541)

karentnarvaez committed:
Sortable Bid Adapter: add eids support (#6565)

* Add Sortable adapter for Prebid 3.x

Update tests to reflect changes.

* Add .js in imports

* hostname not host: don't include port

* Trivial change to trigger build: failure wasn't our adapter

* More failures in other adapters

* PR Feedback

- use https for URL
- fix examples in markdown
- request to endpoint should work now

* Feedback: add native and video examples

* Update unit tests

Co-authored-by: Shannon Broekhoven <shannon@sortable.com>

Commit:
Outbrain Bid Adapter: replacing Zemanta (#6558)

nouchy committed:
Sirdata Real-time Data Module: add new RTD module (#6515)

logicad committed:
Logicad Bid Adapter: add support for userid modules (#6529)
